packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -445,14 +449,36 @@ class Scrollable extends StatefulWidget {
     RenderObject? targetRenderObject;
     ScrollableState? scrollable = Scrollable.maybeOf(context);
     while (scrollable != null) {
-      futures.add(scrollable.position.ensureVisible(
-        context.findRenderObject()!,
-        alignment: alignment,
-        duration: duration,
-        curve: curve,
-        alignmentPolicy: alignmentPolicy,
-        targetRenderObject: targetRenderObject,
-      ));
+      if (scrollable is _HorizontalInnerDimensionState) {
```

Comment:
Yeah I know... I don't love it but have not found a way around it yet. I thought since it is private API it might qualify for an exception to the `is` check.

> here, make this thing visible and it internally can then dispatch to its horizontal and vertical position?

How could we identify that special case? 2D scrolling does not use custom ScrollPosition classes.

Reply:
> How could we identify that special case? 2D scrolling does not use custom ScrollPosition classes.

The scrollable knows whether it is part of 2D scrolling or not, right? I though, below you could just call something like `scrollable.performEnsureVisible` (better name TBD) and then the scrollable would internally dispatch to its vertical and horizontal position if it is a 2d scroller or its single position if it is a 1d scroller. `performEnsureVisible` the probably needs to return some kind of a composited return type that includes the next scrollable where to continue the search since it needs to jump over one.

Reply:
Ah I see! Thank you! I clearly got stuck in my thinking here. I will update this week!